httsleep/main.py

```diff
@@ -12,6 +12,7 @@
 DEFAULT_POLLING_INTERVAL = 2 # in seconds
 DEFAULT_MAX_RETRIES = 50
 VALID_CONDITIONS = ['status_code', 'json', 'jsonpath', 'text', 'callback']
+DEFAULT_SESSION = requests.Session()
```

Comment:
Since our `PreparedRequest` now comes from [`self.session.prepare_request(self.request)`](https://github.com/requests/requests/blob/master/requests/models.py#L248-L260) instead of [`self.request.prepare()`](https://github.com/requests/requests/blob/master/requests/sessions.py#L426-L438) - see below - there is **a behavioural change in the default case (no `Session` explicitly provided)**.
    -  From the links above, you can see that this involves merging in `Session` default headers, cookies, params, auth, and hooks.  Of these, `params` and `auth` default to empty / none; I think the only interesting possibilities are around `cookies` and `headers`.
    -  If we consider the behavioural change acceptable, then we're good.  If on the other hand we want to ensure no behavioural change, we can change the definition of `DEFAULT_SESSION` to override cookies and headers to empty dictionaries.  Your call.
    -  Additionally (Python style question) should `DEFAULT_SESSION` be a method rather than an instance variable?  I worry that the lack of immutability means that somebody could accidentally change `DEFAULT_SESSION` and unintentionally affect other uses of it.